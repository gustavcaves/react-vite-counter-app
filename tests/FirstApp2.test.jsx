import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en <FirstApp />', () => {

    const title = "Hola, Soy Gokú";
    const subTitle = 'Soy un subtitulo'


    test('debe hacer match con el snapshopt', () => {
        const {container } = render(<FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, Soy Gokú', () => {
        
        render(<FirstApp title={title}/>)
        expect( screen.getByText(title) ).toBeTruthy(); 
        // expect( screen.getByText(title) ).not.toBeTruthy(); 
        // screen.debug();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title}/>)
        expect( screen.getByRole('heading', { level: 1}).innerHTML).toContain(title)
        // console.log(screen.getByRole('heading', { level: 1}).innerHTML)
    });


    test('debe de mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp 
                title={title}
                subTitle={subTitle}
            />
        );
        expect( screen.getAllByText(subTitle).length).toBe(2)
        
    });

});