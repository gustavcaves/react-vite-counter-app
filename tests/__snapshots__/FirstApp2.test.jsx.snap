// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshopt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Gokú
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
